src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/DeviceAuthHelper.cs

```diff
@@ -77,11 +75,99 @@ public static string GetBypassChallengeResponse(Dictionary<string, string> respo
         //This corresponds to windows 7, 8, 8.1 and their server equivalents.       
         public static bool CanOSPerformPKeyAuth()
         {
-#if NET_CORE || NET5_WIN || DESKTOP
+#if NET_CORE || NET5_WIN || DESKTOP || NETSTANDARD
             return !Platforms.Features.DesktopOs.DesktopOsHelper.IsWin10OrServerEquivalent();
 #else
             return false;
 #endif
         }
+
+        /// <summary>
+        /// Formats PKeyAuth header
+        /// </summary>
+        public static void FormatResponseHeader(
+            DeviceAuthJWTResponse responseJWT,
+            byte[] signedResponse,
+            IDictionary<string, string> challengeData,
+            out string responseHeader)
+        {
+            string authHeaderTemplate = "PKeyAuth {0}, Context=\"{1}\", Version=\"{2}\"";
+
+            string signedJwt = string.Format(CultureInfo.InvariantCulture, "{0}.{1}", responseJWT.GetResponseToSign(), Base64UrlHelpers.Encode(signedResponse));
+            string authToken = string.Format(CultureInfo.InvariantCulture, " AuthToken=\"{0}\"", signedJwt);
```

Comment:
question: can we do string interpolation here? it's like 5-10x faster than string.Format.

Reply:
Done.